apps/studio/src/features/editing-experience/components/PreviewIframe.tsx

```diff
@@ -0,0 +1,72 @@
+import type { PropsWithChildren } from "react"
+import { useEffect, useState } from "react"
+import { Flex } from "@chakra-ui/react"
+import Frame, { useFrame } from "react-frame-component"
+
+export const PreviewIframe = ({ children }: PropsWithChildren): JSX.Element => {
+  // TODO: Add toolbar for users to adjust the width of the iframe
+  const [width, _setWidth] = useState("100%")
+  return (
+    <Flex
+      borderRadius="8px"
+      bg="white"
+      shadow="md"
+      justify="center"
+      overflow="auto"
+      w={width}
+      h="100%"
+    >
+      <Frame
+        style={{
+          // TODO: Inject CSS variables for site theme here.
+          width,
+        }}
+        head={
+          // eslint-disable-next-line @next/next/no-css-tags
+          <link
+            rel="stylesheet"
+            type="text/css"
+            href="/assets/css/preview-tw.css"
+          />
+        }
+      >
+        <IframeInnerComponent>{children}</IframeInnerComponent>
+      </Frame>
+    </Flex>
+  )
+}
+
+const IframeInnerComponent = ({ children }: PropsWithChildren) => {
+  const { document: iframeDocument } = useFrame()
+
+  // !! This effect might break usages of scroll lock if scroll lock is not triggered by inside the iframe.
+  useEffect(() => {
+    // Frustratingly, rendering react in an iframe still applies the styles to the parent document instead of the iframe
+    // This effect synchronise any changes to the document's styles with the iframe.
```

Comment:
styles are applying to root document instead of the iframe for reasons i do not want to dig into now haha. So just use this way for now to remove styles applied onto root document's `head` and put it into iframe's `head`